activerecord/lib/active_record/fixture_set/file.rb

```diff
@@ -18,12 +18,25 @@ def self.open(file)
       def initialize(file)
         @file = file
         @rows = nil
+        @model_class = nil
       end
 
       def each(&block)
         rows.each(&block)
       end
 
+      def model_class
+        return @model_class if @model_class
+
+        rows.delete_if do |fixture_name, row|
```

Comment:
This implementation is based on the fact that `model_class` will be called before `each`, otherwise you'd get a fixture for `_fixture`. We should probably change it to behave consistently regardless of call order. Meaning that we have to treat the `_fixture` key special that it won't ever be part of rows.

This could look something like this (pseudocode):

```ruby
def model_class
  raw_content["_fixture"]
  # ...
end

def rows
  raw_content.except "_fixture"
end

def raw_content
  # read yaml-file and cache contents
end
```